kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/matchers.kt

```diff
@@ -4,13 +4,16 @@ package io.kotlintest.matchers
 
 import io.kotlintest.*
 
-fun withClue(clue: String, thunk: () -> Any) {
-  val currentClue = ErrorCollector.clueContext.get()
+fun <ReturnType> withClue(clue: Any, thunk: () -> ReturnType): ReturnType {
+    return clue.asClue { thunk() }
+}
+
+fun <ClueType, ReturnType> ClueType.asClue(body: (ClueType) -> ReturnType): ReturnType {
```

Comment:
Could you please add some javadocs to this method? It may improve usability when users try to verify what this does inside the IDE (CTRL + Q).

Reply:
Will do!